Tests/BackendIntegrationTests/BaseBackendIntegrationTests.swift

```diff
@@ -128,12 +124,26 @@ private extension BaseBackendIntegrationTests {
         }
     }
 
+    func createUserDefaults() {
+        self.userDefaults = UserDefaults(suiteName: Constants.userDefaultsSuiteName)
+        self.userDefaults.removePersistentDomain(forName: Constants.userDefaultsSuiteName)
+        self.userDefaults.synchronize()
```

Comment:
let's add a comment here in case a future dev wonders why this is needed, reads Apple's docs and decides to remove it as "not needed"

<img width="1097" alt="Screenshot 2023-06-27 at 2 54 02 PM" src="https://github.com/RevenueCat/purchases-ios/assets/3922667/b9c47c7c-4a44-43fb-8908-90f13f1f4220">

Reply:
👍🏻 
That's also why I added the assertion below, but I'll add a comment too.